### PR TITLE
Minor changes

### PR DESCRIPTION
**Pull request by Ngoguey42**
- Update many docstrings
- Fix test_data_source_modes.py
- Fix DataSource.activate_all

### CODE REVIEW

File: buzzard/_a_emissary.py [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
 class AEmissary(AStored):
-    """Stored proxy that has a driver notion"""
+    """Base abstract class defining the common behavior of all sources that are backed by a driver.
+
+    Features Defined
+    ----------------
+    - Has a `driver` (like "GTiff" for GDAL's geotiff driver)
+    - Has `open_options`
+    - Has a `path` (if the driver supports it)
+    - Can deleted (if the driver supports it)
```
Comment (azarz) [inline buzzard/_a_emissary.py]: `Can be deleted`

File: buzzard/_a_pooled_emissary.py [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+
+    Features Defined
+    ----------------
+    - An `activate` method to manually open the driver (Mostly useless feature since opening
+        is automatic if necessary)
+    - A `deactivate` method to close the driver (Useful to flush data to disk)
+    - An `active_count` property
+    - An `active` property
+    """
+    # """Emissary that has a notion of activation/deactivation (file descriptor)"""
```
Comment (azarz) [inline buzzard/_a_pooled_emissary.py]: Delete this line

File: buzzard/_a_proxy_vector.py [hunk 3]
```diff
@@ -11,7 +11,17 @@
 from buzzard._tools import conv
 
 class AProxyVector(AProxy):
-    """Base abstract class defining the common behavior of all vectors"""
+    """Base abstract class defining the common behavior of all vectors.
+
+    Features Defined
+    ----------------
+    - Has a `type` that defines the type of geometry (like "Polygon")
+    - Has `fields` that defines the type of informations that are paired with each geometries
```
Comment (azarz) [inline buzzard/_a_proxy_vector.py]: ```Has `fields` that define the type of information that is paired with each geometries```

File: buzzard/_a_emissary.py (hunk 1 above)
Comment (Ngoguey42) [reply buzzard/_a_emissary.py]: Done

File: buzzard/_a_pooled_emissary.py (hunk 2 above)
Comment (Ngoguey42) [reply buzzard/_a_pooled_emissary.py]: Done

File: buzzard/_a_proxy_vector.py (hunk 3 above)
Comment (Ngoguey42) [reply buzzard/_a_proxy_vector.py]: Done

File: buzzard/_a_proxy_vector.py [hunk 4]
```diff
@@ -11,7 +11,17 @@
...
+    - Has a `stored` extent that allows to retreive the current extent of all the geometries
+    - Has a length that indicates how many geometries this source contains.
+    - Has several read functions (like `iter_data`) to retreive geometries in their current state to
```
[the remark this answers is not in the document]
Comment (Ngoguey42) [reply buzzard/_a_proxy_vector.py]: Done

File: buzzard/_a_proxy_vector.py [hunk 5]
```diff
@@ -11,7 +11,17 @@
...
 
 class AProxyVector(AProxy):
-    """Base abstract class defining the common behavior of all vectors"""
+    """Base abstract class defining the common behavior of all vectors.
+
+    Features Defined
+    ----------------
+    - Has a `type` that defines the type of geometry (like "Polygon")
+    - Has `fields` that defines the type of informations that are paired with each geometries
+    - Has a `stored` extent that allows to retreive the current extent of all the geometries
```
[the remark this answers is not in the document]
Comment (Ngoguey42) [reply buzzard/_a_proxy_vector.py]: Done

File: buzzard/_datasource.py [hunk 6]
```diff
@@ -858,13 +885,26 @@ def activate_all(self):
...
+            if isinstance(prox, APooledEmissary)
+        ]
+        total = len(proxs)
+
+        if self._back.max_active < total:
+            raise RuntimeError("Can't activate all pooled sources at the same time: {} pooled sources and max_activated is {}".format(
+                total, self._back.max_active,
             ))
-        for prox in self._keys_of_proxy.keys():
+        i = 0
```
[the remark this answers is not in the document]
Comment (Ngoguey42) [reply buzzard/_datasource.py]: sure

**Pull request merged**